scripts/validate-notebooks.ps1

```diff
@@ -1,51 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
 # Script to validate Jupyter notebooks.
+$all_ok = $True
 
-#
 # This function takes a folder with Katas. Copies the corresponding 
 # jupyter notebook into a "Check.ipynb" that replaces the %kata magic
 # with a %check_kata magic that runs the kata not with the user-supplied code
 # but with the Reference implementation.
 # If the execution fails or has warnings, the execution of the notebook
 # will fail.
 function validate {
-    Param($folder)
-    pushd $folder
+    Param($Notebook)
 
     # Name of the notebook that will be used for checking katas.
     $CheckNotebook = 'Check.ipynb'
 
+    $folder = Resolve-Path($Notebook.Directory)
+    pushd $folder
+
+
     # Make sure old versions are removed.
     if (Test-Path $CheckNotebook)  {
         Remove-Item $CheckNotebook
     }
 
     # Find the name of the kata's notebook.
-    $Notebook = (Get-ChildItem *.ipynb)
     Write-Host "Checking notebook $Notebook."
 
     #  convert %kata to %check_kata. run Jupyter nbconvert to execute the kata.
-    (Get-Content $Notebook -Raw) |  ForEach-Object { $_.replace('%kata', '%check_kata') } | Set-Content $CheckNotebook -NoNewline
+    (Get-Content $Notebook -Raw) | ForEach-Object { $_.replace('%kata', '%check_kata') } | Set-Content $CheckNotebook -NoNewline
     jupyter nbconvert $CheckNotebook --execute  --ExecutePreprocessor.timeout=120
 
     # if jupyter returns an error code, return that this notebook is invalid:
-    $invalid = ($LastExitCode -ne 0)
-    Write-Host "Result: " $LastExitCode
+    if ($LastExitCode -ne 0) {
+        Write-Host "##vso[task.logissue type=error;]Validation errors for $Notebook ."        
+        $script:all_ok = $false
+    }
 
     popd
-    return $invalid
 }
 
-# List of Katas to verify:
+# List of Notebooks that can't be validated as they required user-input
+# other than the kata answer:
+$not_ready = 
+@(
+    'Check.ipynb',
+    'CHSHGame.ipynb',
+    'DeutschJozsaAlgorithm.ipynb',
+    'GHZGame.ipynb',
```

Comment:
What are the conditions that put a kata in this list? Looking at `GHZGame`, it does not seem to have anything odd going on with it (like open-ended tasks or using a different simulator). Is it that one of the tests depends on solution to one of the previous tasks? That should have been in https://github.com/microsoft/QuantumKatas/pull/116.

It would be really nice to document this a bit.

Reply:
The condition was: did it ran successfully? In general, I didn't took the time to figure out why it failed. I know that for DJ the problem is that simulate uses an operation defined in a Kata, which is currently not supported (we would need to do a similar trick to what we did for the games), for the others I suspect something similar.

Reply:
I spent some time investigating, fixed a couple of katas and documented the ones I couldn't fix in https://github.com/microsoft/QuantumKatas/pull/159.